src/figuresClasses.ts

```diff
@@ -1,11 +1,84 @@
-export interface Figure {}
+import { roundedDown } from './utility/roundedDown';
 
-export class Triangle implements Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Circle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => {};
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
```

Comment:
[CODE KNOWLEDGE] - You don't need to manually assign values to class properties when using access modifiers in the constructor parameters. TypeScript does this automatically.